src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchResultJsonRenderer.cs

```diff
@@ -16,32 +13,30 @@ internal class PackageSearchResultJsonRenderer : IPackageSearchResultRenderer
     {
         private ILoggerWithColor _logger;
         private PackageSearchVerbosity _verbosity;
+        private bool _exactMatch;
         private PackageSearchMainOutput _packageSearchMainOutput;
 
-        public PackageSearchResultJsonRenderer(ILoggerWithColor loggerWithColor, PackageSearchVerbosity verbosity)
+        public PackageSearchResultJsonRenderer(ILoggerWithColor loggerWithColor, PackageSearchVerbosity verbosity, bool exactMatch)
         {
             _logger = loggerWithColor;
             _verbosity = verbosity;
+            _exactMatch = exactMatch;
         }
         public async void Add(PackageSource source, IEnumerable<IPackageSearchMetadata> completedSearch)
         {
             PackageSearchResult packageSearchResult = new PackageSearchResult(source.Name);
 
             foreach (IPackageSearchMetadata metadata in completedSearch)
             {
-                if (_verbosity == PackageSearchVerbosity.Minimal)
-                {
-                    packageSearchResult.Packages.Add(new MinimalVerbosityPackage(metadata));
-                }
-                else if (_verbosity == PackageSearchVerbosity.Detailed)
+                string deprecation = "";
+
+                if (_verbosity == PackageSearchVerbosity.Detailed)
                 {
                     PackageDeprecationMetadata packageDeprecationMetadata = await metadata.GetDeprecationMetadataAsync();
```

Comment:
Looks like we are making the network call here to `metadata.GetDeprecationMetadataAsync()` and we are not using the method result. If we can somehow address this comment, then I guess we don't need to make another blocking call in the new `SearchResultPackagesConverter` code. Please let me know If I am missing something.

Reply:
https://github.com/NuGet/NuGet.Client/pull/5602#discussion_r1491862567 
Previously I had it so that the aync call to `metadata.GetDeprecationMetadataAsync() `  is called in the `PackageSearchResultJsonRenderer.cs`. However, that mean I had to create a class containing both `IPackageSearchMetadata` and deprecation message for each package. Instead of that, the new implementation does a blocking call to `GetDeprecationMetadataAsync()` . That means we can remove the new class I had created to represent packages.

Reply:
`.Result` isn't necessarily bad in this situation. 
The API probably isn't the best. There's never any actual calculation happening in the async method.
There's also no deadlock concerns here.